pkg/upgrade/upgrade.go

```diff
@@ -275,6 +276,12 @@ func CleanupExistingResource(ctx context.Context,
 			return err
 		}
 	}
+	// remove modelreg proxy container from deployment in ODH
+	if platform == cluster.OpenDataHub {
```

Comment:
Is this only needed for ODH? @dhirajsb

Reply:
need  @dhirajsb to confirm, but when i check rhoai-2.14 2.15 i can see the deployment had been updated.

Reply:
yes, confirmed it's only needed for odh